server/chat-plugins/chat-monitor.ts

```diff
@@ -305,15 +305,15 @@ export const chatfilter: ChatFilter = function (message, user, room) {
 		.replace(/\u039d/g, 'e');
 	lcMessage = lcMessage.replace(/__|\*\*|``|\[\[|\]\]/g, '');
 
-	const isStaffRoom = room && ((room.chatRoomData && room.roomid.endsWith('staff')) || room.roomid.startsWith('help-'));
+	const isStaffRoom = room && ((room.settings && room.roomid.endsWith('staff')) || room.roomid.startsWith('help-'));
```

Comment:
```suggestion
	const isStaffRoom = room && ((room.settings.persistSettings && room.roomid.endsWith('staff')) || room.roomid.startsWith('help-'));
```

What was your regex? You should be grepping for `\.settings[^.]` because it's nearly definitely a bug.